apps/lib/opt.c

```diff
@@ -356,21 +356,25 @@ void print_format_error(int format, unsigned long flags)
 }
 
 /* Parse a cipher name, put it in *EVP_CIPHER; return 0 on failure, else 1. */
-int opt_cipher(const char *name, const EVP_CIPHER **cipherp)
+int opt_cipher(const char *name, EVP_CIPHER **cipherp)
 {
-    *cipherp = EVP_get_cipherbyname(name);
+    *cipherp = (EVP_CIPHER *)EVP_get_cipherbyname(name);
     if (*cipherp != NULL)
         return 1;
+    *cipherp = EVP_CIPHER_fetch(NULL, name, NULL);
```

Comment:
You're not checking whether `*cipherp` is NULL, or most importantly, not.

Also, I think we concluded at some point that fetching should be done first, with legacy working as a fallback.

Reply:
Added NULL check, and fixed the order to be fetch then legacy.